From 258bf47c4740bb5290c60313595c42ae21d135b5 Mon Sep 17 00:00:00 2001
From: Duncan Macleod <duncan.macleod@ligo.org>
Date: Fri, 14 Jan 2022 11:50:32 +0000
Subject: [PATCH 06/12] python: link against libcondor_utils and libclassad as
 external

don't use the CMake targets, so that we don't have to rebuild these libraries when building the Python bindings
---
 src/python-bindings/CMakeLists.txt | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/src/python-bindings/CMakeLists.txt b/src/python-bindings/CMakeLists.txt
index 30e2e67a55..8562972974 100644
--- a/src/python-bindings/CMakeLists.txt
+++ b/src/python-bindings/CMakeLists.txt
@@ -1,3 +1,6 @@
+get_target_property(UTILS_LIB_NAME condor_utils OUTPUT_NAME)
+find_library(CLASSAD_LIBRARY classad)
+
 if ((WITH_PYTHON_BINDINGS_V2) AND (NOT WINDOWS))
   set( WHEEL_BINARY_DIR "${CMAKE_BINARY_DIR}/bindings/python" )
 
@@ -9,14 +12,14 @@ if ((WITH_PYTHON_BINDINGS_V2) AND (NOT WINDOWS))
 
   add_library(                  py3classad2_impl SHARED classad2_impl.cpp )
   target_include_directories(   py3classad2_impl PUBLIC ${PYTHON3_INCLUDE_DIRS} )
-  target_link_libraries(        py3classad2_impl classad condor_utils ${MAGIC_LINKER_FLAG} )
+  target_link_libraries(        py3classad2_impl ${CLASSAD_LIBRARY} ${UTILS_LIB_NAME} ${MAGIC_LINKER_FLAG} )
   set_target_properties(        py3classad2_impl PROPERTIES SUFFIX ${MVI_SUFFIX} )
   set_target_properties(        py3classad2_impl PROPERTIES PREFIX "" OUTPUT_NAME classad2_impl )
   set_target_properties(        py3classad2_impl PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${WHEEL_BINARY_DIR}/classad2" )
 
   add_library(                  py3htcondor2_impl SHARED htcondor2_impl.cpp )
   target_include_directories(   py3htcondor2_impl PUBLIC ${PYTHON3_INCLUDE_DIRS} )
-  target_link_libraries(        py3htcondor2_impl classad condor_utils ${MAGIC_LINKER_FLAG} )
+  target_link_libraries(        py3htcondor2_impl ${CLASSAD_LIBRARY} ${UTILS_LIB_NAME} ${MAGIC_LINKER_FLAG} )
   set_target_properties(        py3htcondor2_impl PROPERTIES SUFFIX ${MVI_SUFFIX} )
   set_target_properties(        py3htcondor2_impl PROPERTIES PREFIX "" OUTPUT_NAME htcondor2_impl )
   set_target_properties(        py3htcondor2_impl PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${WHEEL_BINARY_DIR}/htcondor2" )
@@ -30,7 +33,7 @@ endif()
 if ((WITH_PYTHON_BINDINGS_V3) AND (NOT WINDOWS))
   add_library(                  py3classad3_impl SHARED classad3_impl.cpp )
   target_include_directories(   py3classad3_impl PUBLIC ${PYTHON3_INCLUDE_DIRS} )
-  target_link_libraries(        py3classad3_impl classad condor_utils ${PYTHON3_LIBRARIES} )
+  target_link_libraries(        py3classad3_impl ${CLASSAD_LIBRARY} ${UTILS_LIB_NAME} ${PYTHON3_LIBRARIES} )
   set_target_properties(        py3classad3_impl PROPERTIES PREFIX "" OUTPUT_NAME classad3_impl )
   set_target_properties(        py3classad3_impl PROPERTIES SUFFIX "${PYTHON3_MODULE_SUFFIX}" )
   install ( TARGETS py3classad3_impl DESTINATION ${C_PYTHON3ARCHLIB}/classad3 )
@@ -47,14 +50,14 @@ if ((WITH_PYTHON_BINDINGS_V2) AND (WINDOWS))
 
   add_library(                  py3classad2_impl SHARED classad2_impl.cpp )
   target_include_directories(   py3classad2_impl PUBLIC ${PYTHON3_INCLUDE_DIRS} )
-  target_link_libraries(        py3classad2_impl ${CLASSADS_FOUND} condor_utils ${PLFN})
+  target_link_libraries(        py3classad2_impl ${CLASSAD_LIBRARY} ${UTILS_LIB_NAME} ${PLFN})
   set_target_properties(        py3classad2_impl PROPERTIES PREFIX "" OUTPUT_NAME classad2_impl )
   set_target_properties(        py3classad2_impl PROPERTIES PREFIX "" RUNTIME_OUTPUT_NAME classad2_impl PDB_NAME py3classad2 )
   set_target_properties(        py3classad2_impl PROPERTIES SUFFIX "${PYTHON3_DLL_SUFFIX}" )
 
   add_library(                  py3htcondor2_impl SHARED htcondor2_impl.cpp )
   target_include_directories(   py3htcondor2_impl PUBLIC ${PYTHON3_INCLUDE_DIRS} )
-  target_link_libraries(        py3htcondor2_impl ${CLASSADS_FOUND} condor_utils ${PLFN})
+  target_link_libraries(        py3htcondor2_impl ${CLASSAD_LIBRARY} ${UTILS_LIB_NAME} ${PLFN})
   set_target_properties(        py3htcondor2_impl PROPERTIES PREFIX "" OUTPUT_NAME htcondor2_impl )
   set_target_properties(        py3htcondor2_impl PROPERTIES PREFIX "" RUNTIME_OUTPUT_NAME htcondor2_impl PDB_NAME py3htcondor2 )
   set_target_properties(        py3htcondor2_impl PROPERTIES SUFFIX "${PYTHON3_DLL_SUFFIX}" )
@@ -66,7 +69,7 @@ endif()
 if ((WITH_PYTHON_BINDINGS_V3) AND (WINDOWS))
   add_library(                  py3classad3_impl SHARED classad3_impl.cpp )
   target_include_directories(   py3classad3_impl PUBLIC ${PYTHON3_INCLUDE_DIRS} )
-  target_link_libraries(        py3classad3_impl ${CLASSADS_FOUND} condor_utils ${PLFN})
+  target_link_libraries(        py3classad3_impl ${CLASSAD_LIBRARY} ${UTILS_LIB_NAME} ${PLFN})
   set_target_properties(        py3classad3_impl PROPERTIES PREFIX "" OUTPUT_NAME classad3_impl )
   set_target_properties(        py3classad3_impl PROPERTIES PREFIX "" RUNTIME_OUTPUT_NAME classad3_impl PDB_NAME py3classad3 )
   set_target_properties(        py3classad3_impl PROPERTIES SUFFIX "${PYTHON3_DLL_SUFFIX}" )
-- 
2.39.5

